// service definition.
syntax = "proto3";
option go_package = ".;option";

service OptionPricing {
  // Method to compute option price
  rpc ComputePrice (ComputeRequest) returns (UxtSlice) {}
}

// The request message containing params
message ComputeRequest {
  double maxPrice = 1;
  double volatility = 2;
  double r = 3;
  double tMax = 4;
  double strikePrice = 5;
  string calculationType = 6;
  double beta = 7;


}

// The response message containing the greetings
message UxtSlice {
  repeated Uxt U = 1;
}

message Uxt {
  repeated double Ut = 1;
}